add-chain/addresses.go

```diff
@@ -15,154 +15,89 @@ type AddressData struct {
 	Address string `json:"address"`
 }
 
-var (
-	// Addresses to retrieve from JSON
-	AddressManager                    = "AddressManager"
-	L1CrossDomainMessengerProxy       = "L1CrossDomainMessengerProxy"
-	L1ERC721BridgeProxy               = "L1ERC721BridgeProxy"
-	L1StandardBridgeProxy             = "L1StandardBridgeProxy"
-	OptimismMintableERC20FactoryProxy = "OptimismMintableERC20FactoryProxy"
-	SystemConfigProxy                 = "SystemConfigProxy"
-	OptimismPortalProxy               = "OptimismPortalProxy"
-	ProxyAdmin                        = "ProxyAdmin"
-
-	// Addresses to retrieve from chain
-	SuperchainConfig  = "SuperchainConfig"
-	Guardian          = "Guardian"
-	Challenger        = "Challenger"
-	ProxyAdminOwner   = "ProxyAdminOwner"
-	SystemConfigOwner = "SystemConfigOwner"
-	Proposer          = "Proposer"
-	UnsafeBlockSigner = "UnsafeBlockSigner"
-	BatchSubmitter    = "BatchSubmitter"
-
-	// Non Fault Proof contracts
-	L2OutputOracleProxy = "L2OutputOracleProxy"
-
-	// Fault Proof contracts:
-	AnchorStateRegistryProxy = "AnchorStateRegistryProxy"
-	DelayedWETHProxy         = "DelayedWETHProxy"
-	DisputeGameFactoryProxy  = "DisputeGameFactoryProxy"
-	FaultDisputeGame         = "FaultDisputeGame"
-	MIPS                     = "MIPS"
-	PermissionedDisputeGame  = "PermissionedDisputeGame"
-	PreimageOracle           = "PreimageOracle"
-)
-
-func readAddressesFromChain(addresses map[string]string, l1RpcUrl string, isFaultProofs bool) error {
+func readAddressesFromChain(addresses *superchain.AddressList, l1RpcUrl string, isFaultProofs bool) error {
 	// SuperchainConfig
-	address, err := castCall(addresses[OptimismPortalProxy], "superchainConfig()(address)", l1RpcUrl)
+	address, err := castCall(addresses.OptimismPortalProxy.String(), "superchainConfig()(address)", l1RpcUrl)
 	if err != nil {
-		addresses[SuperchainConfig] = ""
+		addresses.SuperchainConfig = superchain.Address{}
```

Comment:
the default value is already a zero array, no need to set it explicitly

Reply:
https://github.com/ethereum-optimism/superchain-registry/pull/425/commits/0b48204ccb9226ea7a2a1e0442db21a5f3d54ca3